internal/pkg/file/file.go

```diff
@@ -42,8 +43,11 @@ func Exists(fileName string) bool {
 // Chown changes file/dir mode
 func Chown(file, owner string, permissions os.FileMode) error {
 	// Chown the file properly for the owner
-	uid, _ := users.GetUID(owner)
-	err := os.Chown(file, uid, -1)
+	uid, err := users.GetUID(owner)
+	if err != nil {
+		logrus.Warnf("failed to get uid for %s, using 0: %v", owner, err)
```

Comment:
Good catch. I wonder if it would be better to error out here? The desire to log a warning in a core utility function somehow indicates to me that it's probably a real error condition. You write in your commit message that it's correct to fallback to UID 0. It's not immediately obvious to me why that is. An alternative approach might be to return some well known error that callers can check against and implement their own fallback handling, if they desire to do so.

I'd ask for at least two things:

1. Add an explanatory comment as to why this is desired, maybe directly in the functions godoc string?
2. Explicitly hardcode the UID 0 in the subsequent call to chown, or explicitly set the uid variable to 0 here, instead of implicitly relying on the behavior of GetUID to return 0 on error.